File: app/src/main/java/test/itexico/movies/managers/StandardRequest.java
package test.itexico.movies.managers;

import com.android.volley.Response;
import com.android.volley.toolbox.JsonArrayRequest;

import org.json.JSONArray;

import java.util.HashMap;
import java.util.Map;

import test.itexico.movies.utils.Trakt;

public class StandardRequest extends JsonArrayRequest {

    public StandardRequest(int method, String url, JSONArray jsonRequest, Response.Listener<JSONArray> listener, Response.ErrorListener errorListener) {
        super(method, url, jsonRequest, listener, errorListener);
    }

    @Override
    public Map<String, String> getHeaders() {
        Map<String, String>  headers = new HashMap<>();
        headers.put("Content-Type", "application/json");
        headers.put("trackt-api-version", "2");
        headers.put("trackt-api-key", Trakt.getClientID());
        return headers;
    }
}

File: app/src/main/java/test/itexico/movies/managers/StandardRequest.kt
package test.itexico.movies.managers

import com.android.volley.Response
import com.android.volley.toolbox.JsonArrayRequest
import org.json.JSONArray
import test.itexico.movies.utils.Trakt
import java.util.*

class StandardRequest(method: Int,
                      url: String,
                      jsonRequest: JSONArray? = null,
                      successlistener: Response.Listener<JSONArray>,
                      errorListener: Response.ErrorListener) :
        JsonArrayRequest(method, url, jsonRequest, successlistener, errorListener) {

    override fun getHeaders(): Map<String, String> {
        val headers = HashMap<String, String>()
        headers["Content-Type"] = "application/json"
        headers["trackt-api-version"] = "2"
        headers["trackt-api-key"] = Trakt.clientID
        return headers
    }
}
